git/options.go

```diff
@@ -27,36 +27,49 @@ const (
 	DefaultPublicKeyAuthUser = "git"
 )
 
-// CheckoutOptions are the options used for a Git checkout.
-type CheckoutOptions struct {
-	// Branch to checkout, can be combined with Branch with some
-	// Implementations.
-	Branch string
-
-	// Tag to checkout, takes precedence over Branch.
-	Tag string
-
-	// SemVer tag expression to checkout, takes precedence over Tag.
-	SemVer string `json:"semver,omitempty"`
-
-	// Commit SHA1 to checkout, takes precedence over Tag and SemVer,
-	// can be combined with Branch with some Implementations.
-	Commit string
+// CloneOptions are the options used for a Git clone.
+type CloneOptions struct {
+	// CheckoutStrategy defines a strategy to use while checking out
+	// the cloned repository to a specific target.
+	CheckoutStrategy
 
 	// RecurseSubmodules defines if submodules should be checked out,
 	// not supported by all Implementations.
 	RecurseSubmodules bool
 
-	// LastRevision holds the revision observed on the last successful
-	// reconciliation.
-	// It is used to skip clone operations when no changes were detected.
-	LastRevision string
+	// LastObservedCommit holds the last observed commit hash of a
+	// Git repository.
+	// If provided, the clone operation will compare it with the HEAD commit
+	// of the branch or tag (as configured via CheckoutStrategy) in the remote
+	// repository. If they match, cloning will be skipped and a "non-concrete"
+	// commit will be returned, which can be verified using `IsConcreteCommit()`.
+
```

Comment:
This line break doesn't work well with godoc. The actual comment above doesn't show up in the godoc.